src/helics/core/TimeoutMonitor.h

```diff
@@ -38,7 +39,11 @@ class TimeoutMonitor {
     /** ping all a brokers sub connections*/
     void pingSub(CoreBroker* brk);
     /** set the parent id*/
-    void setParentId(global_broker_id parent_id) { parentConnection.connection = parent_id; };
+    void setParentId(global_broker_id parent_id, bool disable_ping = false)
```

Comment:
This is probably the correct thing to do,  It can't change at present, but as we get a few more steps down the line, the "pingability" could change during the course of an execution so it probably makes sense to have it as a separate function.  I will make the change later today.

Reply:
the latest commit splits out the functions